config/shared_config_test.go

```diff
@@ -510,6 +510,30 @@ func TestNewSharedConfig(t *testing.T) {
 				UseFIPSEndpoint: aws.FIPSEndpointStateDisabled,
 			},
 		},
+		"defaults mode legacy": {
+			ConfigFilenames:      []string{testConfigFilename},
+			CredentialsFilenames: []string{testCredentialsFilename},
+			Profile:              "autodefaultsmode",
+			Expected: SharedConfig{
+				Profile:      "autodefaultsmode",
+				DefaultsMode: aws.AutoDefaultsMode,
+			},
+		},
```

Comment:
nit test case is labeled as model `legacy`, but expecting `auto` model retrieved from config.